libc/unistd_fsdir.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * Testing unistd.h file/fs/directory related functions
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: 
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+
+#include <unity_fixture.h>
+
+#include "common.h"
+
+
+FILE *fd;
+char buf[50];
+const char filename[] = "unistd_fsdir_file";
+const char directoryname[] = "unistd_fsdir_directory";
+int err;
+char toolongpath[PATH_MAX + 16];
+
+TEST_GROUP(unistd_fsdir);
+
+TEST_SETUP(unistd_fsdir)
+{
+	/* assert we are in root */
```

Comment:
Also, do we want to perform tests in root dir? Maybe /tmp would be better (or some we directory? tmp might have some other fs mounted, so it might not be the best choice I guess)

Reply:
This gives us flexibility across targets as some have only root and `/dev` from the start